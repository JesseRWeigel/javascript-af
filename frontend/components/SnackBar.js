import React, { Component } from 'react'
import styled from 'styled-components'

export const Container = styled.div`
  box-sizing: border-box;
  display: flex;
  position: fixed;
  top: 13%;
  align-items: flex-end;
  right: 2%;
  width: 100vw;
  z-index: 1000;
  flex-direction: column;
`

const Toast = styled.div`
  background-color: #323232;
  margin-bottom: 10px;
  transition: all 0.4s ease-in-out;
  border-radius: 2px;
  display: ${props => (props.show ? 'flex' : 'none')};
  justify-content: space-between;
  max-width: 568px;
  min-width: 288px;
  line-height: 1.5;
  padding: 12px 24px;
`

const Message = styled.p`
  color: #f1f1f1;
`

const Action = styled.a`
  color: #a28aff;
  cursor: pointer;
  margin-left: 24px;
  text-transform: uppercase;
`

class SnackBar extends Component {
  state = {
    show: false
  }

  componentDidMount() {
    this.setState({ show: true }, () => {
      if (this.props.actionText) return
      setTimeout(() => {
        this.setState({ show: false })
      }, this.props.timeout || 4000)
    })
  }

  clickHandler = () => {
    this.props.actionClick && this.props.actionClick()
    this.setState({ show: false })
  }

  render() {
    const { message, actionText } = this.props
    const { show } = this.state
    return (
      <Toast show={show}>
        <Message>{message}</Message>
        {actionText && <Action onClick={this.clickHandler}>{actionText}</Action>}
      </Toast>
    )
  }
}

export default SnackBar
