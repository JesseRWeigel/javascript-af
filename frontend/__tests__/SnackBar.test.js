import React from 'react'
import { render, Simulate } from 'react-testing-library'
import SnackBar from '../components/SnackBar'

describe('SnackBar', () => {
  const actionClick = jest.fn()
  const fullComp = <SnackBar show message="test message" actionText="action" actionClick={actionClick} />
  test('should render with text and action (Snapshot)', () => {
    const wrapper = render(fullComp)
    expect(wrapper.container.firstChild).toMatchSnapshot()
  })
  test('should call onClick when action is clicked', () => {
    Simulate.click(render(fullComp).getByText('action'))
    expect(actionClick).toBeCalled()
  })
})
