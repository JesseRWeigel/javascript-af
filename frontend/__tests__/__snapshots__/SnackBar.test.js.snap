// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnackBar should render with text and action (Snapshot) 1`] = `
<div
  class="SnackBar__Toast-g1j844-1 cSPTBW"
>
  <p
    class="SnackBar__Message-g1j844-2 lgIjyS"
  >
    test message
  </p>
  <a
    class="SnackBar__Action-g1j844-3 egiDpc"
  >
    action
  </a>
</div>
`;
